packages/api/src/external/commonwell/document/document-query.ts

```diff
@@ -115,6 +120,28 @@ export async function queryAndProcessDocuments({
   }
 
   try {
+    const { organization, facility } = await getPatientDataWithSingleFacility(
+      patientParam,
+      facilityId
+    );
+
+    const patientCWData = getCWData(patientParam.data.externalData);
+    const hasNoCWStatus = !patientCWData || !patientCWData.status;
+
+    // If patient has no links in CW then we want to schedule a doc query
+    // and to prevent infinite loops we need to check if the patient has any CQ orgs
+    // as its only called the first time
+    if (hasNoCWStatus && getOrgIdExcludeList) {
```

Comment:
`getOrgIdExcludeList` is not optional, so no need to check that, right?

Shouldn't we also execute this block if there's a status and it's `processing`? Likely with a diff conditional, so:
- no status/link: link (update)
- `processing`: just schedule DQ

Lastly, on the link/update, I'd create a function called `linkPatientToCW` (or similar) that only calls `update`, so when one is reading the code its very clear what's being done - and we don't need comments. :)

Reply:
`getOrgIdExcludeList` is optional missed that when adding it back as a function.

Reply:
The idea is we would only pass that in when we first do a doc query but not from anywhere else

Reply:
The problem is that it creates a situation where we don't have CW data/status and can't execute this block of code, going to the rest of the function which is actual DQ.

Reply:
Can just make it required and pass it from `resetPatientScheduledDocQueryRequestId` - that can get from from `update` which already has it there.